### PR TITLE
Migrate off of enzyme to react testing-library #850

### PR DESCRIPTION
**Pull request by chynh**
# Description of changes

Refactored the following tests to use react-testing-library:
- **SocialLoginGroup.test.js** 
- **MilitaryDetails.test.js** 
- **MilitaryStatus.test.js** 
- **ProfessionalDetails.test.js**
- **Technology.test.js**

# Issue Resolved

Partial fix to #850

### CODE REVIEW

**Comment by kylemh** [comment]
🤖 **This is a bot** 🤖<br /><br />🎉 Deployed Storybook preview! 🎉<br /><br />Click the link at the bottom of this comment to see it live.

Built with commit 97e8cc523df0abc1448a7f0c50c53b28217b8e26

https://deploy-preview-869--operation-code-storybook.netlify.com

File: components/UpdateProfileForm/steps/__tests__/MilitaryDetails.test.js [hunk 1]
```diff
@@ -41,26 +40,24 @@ describe('UpdateProfileForm/Steps/MilitaryDetails', () => {
...
-    wrapper
-      .find('input#yearsOfService')
-      .simulate('change', { target: { id: 'yearsOfService', value: '3' } });
-
-    wrapper.find('input#payGrade').simulate('change', { target: { id: 'payGrade', value: 'E-5' } });
-
-    wrapper.find('form').simulate('submit');
-    await asyncRenderDiff(wrapper);
+    fireEvent.submit(container.querySelector('form'));
+    await wait();
```
Comment (kylemh) [inline components/UpdateProfileForm/steps/__tests__/MilitaryDetails.test.js]:
```suggestion
    await wait(() => {
      expect(OperationCodeAPIMock.history.patch.length).toStrictEqual(1);
    });
```

Comment (kylemh) [reply components/UpdateProfileForm/steps/__tests__/MilitaryDetails.test.js]: This suggestion applies to all the other files you touched I think too!

Comment (chynh) [reply components/UpdateProfileForm/steps/__tests__/MilitaryDetails.test.js]: got it.

**Pull request merged**